src/components/Form/Combobox.js

```diff
@@ -0,0 +1,333 @@
+    labelMultiselect: '',
+    multiEntity: false,
+    onAdd: null,
+    onChange: null,
+    readOnly: false,
+    required: false,
+    size: null,
+    value: '',
+  }
+
+  state = { open: false, inputValue: '' };
+
+  onToggle = () => this.setState(({ open }) => ({ open: !open }))
+
+  onInputKeyDown = ({ keyCode, target: { value } }) => {
+    const { onAdd } = this.props;
+
+    // enter pressed?
+    if (keyCode === 13 && value && onAdd) {
+      onAdd(value);
+    }
+  };
+
+  onSelect(key) {
+    const { closeOnSelect, onSelect } = this.props;
+
+    onSelect(key);
+
+    if (closeOnSelect) this.setState({ open: false });
+  }
+
+  onClickOutside = () => {
+    this.setState({ open: false });
+  }
+
+  getSelectHandler = itemKey => memoize(() => this.onSelect(itemKey));
+
+  getSelectedItems() {
+    const { items } = this.props;
+
+    return items.filter(item => item.selected === true);
+  }
+
+  handleClickOutside() {
+    this.setState({ open: !this.state.open });
+  }
+
+  renderInput() {
+    const {
+      id,
+      disabled,
+      required,
+      labelMultiselect,
+      onChange,
+      placeholder,
+      readOnly,
+      value,
+    } = this.props;
+
+    const formatLabel = () => {
+      if (readOnly) {
+        const selectedItems = this.getSelectedItems();
+        const count = selectedItems.length;
+
+        if (count > 1) {
+          return `${count} ${labelMultiselect}`;
+        } else if (count === 1) {
+          return selectedItems[0].label;
+        }
+      } else {
+        return value;
+      }
+
+      return '';
+    };
+
+    const label = formatLabel();
+
+    const classNames = cx(
+      'slds-combobox__input',
+      { 'slds-combobox__input-value': !!label },
+    );
+
+    const iconRight = readOnly ? 'down' : 'search';
+
+    return (
+      <InputRaw
+        aria-controls={`listbox-${id}`}
+        autoComplete="off"
+        className={classNames}
+        disabled={disabled}
+        iconRight={iconRight}
+        iconRightOnClick={this.onToggle}
+        id={`combobox-${id}`}
+        onChange={onChange}
+        onClick={this.onToggle}
+        onKeyDown={readOnly ? null : this.onInputKeyDown}
+        placeholder={placeholder}
+        readOnly={readOnly}
+        required={required}
+        role="textbox"
+        value={label}
+      />
+    );
+  }
+
+  renderComboboxItems() {
+    const { items } = this.props;
+
+    return items.map(item => (
+      <ComboboxDropdownListItem
+        icon={item.icon}
+        id={`listbox-option-${item.key}`}
+        isHeader={item.isHeader}
+        key={item.key}
+        onClick={this.getSelectHandler(item.key)}
+        selected={item.selected}
+      >
+        {item.label}
+      </ComboboxDropdownListItem>
+    ));
+  }
+
+  renderInlineListbox() {
+    const selectedItems = this.getSelectedItems();
+
+    return (
+      <Listbox className="slds-listbox_inline">
+        {selectedItems.map(({ icon, key, label }) => (
+          <Pill
```

Comment:
`renderInlineListBox` and `renderSelectedPills` are pretty much identical and could be DRYed up with little effort